src/main/java/frc/robot/sensors/IrSensor.java

```diff
@@ -0,0 +1,66 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+/**
+ * Code by Charlie Macdonald.
+ * Last modified on Feb 5th, 2020.
+ */
+
+package frc.robot.sensors;
+
+import edu.wpi.first.wpilibj.AnalogInput;
+import edu.wpi.first.wpilibj.MedianFilter;
+import edu.wpi.first.wpilibj.Sendable;
+import edu.wpi.first.wpilibj.smartdashboard.SendableBuilder;
+import edu.wpi.first.wpilibj.smartdashboard.SendableRegistry;
+
+public class IrSensor implements Sendable {
+    private static final int kIRPort = 1; //set IR sensor port
+        //TODO: The IR sensor's port may change, depending on the robot setup. (Use a robot-specific config to set this?)
+
+    //median filter to discard outliers. Filters over 10 samples
+    private final MedianFilter m_filter = new MedianFilter(10);
+
+    private final AnalogInput m_irsensor;
+
+    //Set the formula values to calculate IR sensor distance:
+        //Values for short range (5-30cm) sensor: 12.8528, -0.007545, -3.92, 35, 4.5.
+        //TODO: If a different IR sensor is used, these values will need to be updated. (Use a robot-specific config to set them?)
+    private final double irSlopeValue = 12.8528;
+    private final double irXValue = -0.007545;
+    private final double irConstant = -3.92;
+    private final double irMaxDistance = 35;
+    private final double irMinDistance = 4.5;
+  
+    public double getDistance() {
+        //Calculate distance (cm) from mV:
+        double currentDistance = ((1/(((m_filter.calculate(m_irsensor.getValue()))/1000 +irXValue)/irSlopeValue))+irConstant);
```

Comment:
This line is pretty hard to read, maybe break it up a bit?
Also, the median filter might not be a good idea unless the sensor is constantly getting values.
For example: if the sensor is polled 20 times at 10cm, and then a minute later is polled once at 30cm the previous 9 polls are being included in this calculation when they are far off from the actual value.

Reply:
We could do two things..

1 - remove the median filter if we think this class will only be used for "ad hoc" readings

2 - add a "takeMeasurement()" method to the class and tell users to put a call to that in the periodic() method of whatever subsystem instantiates one of these.

Reply:
Added a takeMeasurement() method to be called in a periodic method. Updates the median filter and returns the voltage, if anyone wishes to use the latter.